### PR TITLE
Add support for sized primitives

### PR DESCRIPTION
**Pull request by hsyl20**
Fix for sized primitives 

https://gitlab.haskell.org/ghc/ghc/-/issues/19026
https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4560

### CODE REVIEW

File: templates/GenericTemplate.hs [hunk 1]
```diff
@@ -163,13 +164,23 @@ alex_scan_tkn user__ orig_input len input__ s last_acc =
...
       case fromIntegral c of { IBOX(ord_c) ->
         let
+ALEX_IF_GHC_GT_901
+                base   = extendInt32# (alexIndexInt32OffAddr alex_base s)
+                offset = PLUS(base,ord_c)
+                check  = extendInt16# (alexIndexInt16OffAddr alex_check offset)
+
+                new_s = if GTE(offset,ILIT(0)) && EQ(check,ord_c)
+                          then extendInt16# (alexIndexInt16OffAddr alex_table offset)
+                          else extendInt16# (alexIndexInt16OffAddr alex_deflt s)
```
Comment (Ericson2314) [inline templates/GenericTemplate.hs:174]:
```suggestion
                base   = int32ToInt# (alexIndexInt32OffAddr alex_base s)
                offset = PLUS(base,ord_c)
                check  = int16ToInt# (alexIndexInt16OffAddr alex_check offset)

                new_s = if GTE(offset,ILIT(0)) && EQ(check,ord_c)
                          then int16ToInt# (alexIndexInt16OffAddr alex_table offset)
                          else int16ToInt# (alexIndexInt16OffAddr alex_deflt s)
```

**Comment by Ericson2314** [comment]
F.Y.I the change I added is for https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4492. I don't think any other MRs on the horizon are due to affect these primops, so I'm reasonably confident those will be the final form for GHC 9.2.

As discussed elsewhere, eventually I hope to not be confined to hackage releases of Alex and Happy so we don't have to guess what will be in the next version at all, or risk polluting hackage in so guessing (lest the guess be wrong).

**Comment by hsyl20** [comment]
Thanks @Ericson2314!

**Comment by Ericson2314** [comment]
@hsyl20 want  to get on IRC for a second to discuss what we should do next?

**Pull request merged**